cocos/scripting/lua-bindings/manual/CCLuaEngine.cpp

```diff
@@ -439,13 +439,13 @@ int LuaEngine::handleCommonEvent(void* data)
         return 0;
    
     CommonScriptData* commonInfo = static_cast<CommonScriptData*>(data);
-    if (NULL == commonInfo->eventName || 0 == commonInfo->handler)
+    if ('\0' == commonInfo->eventName[0] || 0 == commonInfo->handler)
```

Comment:
Sorry its line 534:

https://github.com/cocos2d/cocos2d-x/blob/c9306a053f051325a03b5297be7be6d645584780/cocos/base/CCScriptSupport.h#L534

Xpol Wan <xpolife@gmail.com>于2016年6月22日周三 下午10:50写道：

> See
> https://github.com/cocos2d/cocos2d-x/blob/c9306a053f051325a03b5297be7be6d645584780/cocos/base/CCScriptSupport.h#L586
>
> CommonScriptData::eventName is array of char not pointer to char. So NULL
> == commonInfo->eventName will always eval to true.
>
> This cimmit just do the intended check.
> Its noting wrong about this fix, it just disclosure a bug hidden in our
> code.
>
> James Chen <notifications@github.com>于2016年6月22日周三 下午5:28写道：
>
>> In cocos/scripting/lua-bindings/manual/CCLuaEngine.cpp
>> <https://github.com/cocos2d/cocos2d-x/pull/15443#discussion_r68021505>:
>>
>> > @@ -439,13 +439,13 @@ int LuaEngine::handleCommonEvent(void* data)
>> >          return 0;
>> >
>> >      CommonScriptData* commonInfo = static_cast<CommonScriptData*>(data);
>> > -    if (NULL == commonInfo->eventName || 0 == commonInfo->handler)
>> > +    if ('\0' == commonInfo->eventName[0] || 0 == commonInfo->handler)
>>
>> What's the warning of this?
>> XMLHttpRequest & WebSocketTest in lua test will not get response after
>> this change.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/cocos2d/cocos2d-x/pull/15443/files/92fe5fa308c98782f7a876336a9efe9460dbe5fa#r68021505>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AABPGAWyhyprqSi_AOXwQKfEZQTagkMUks5qOQBZgaJpZM4IJcAz>
>> .
>>
>